src/paths/random_walk.c

```diff
@@ -132,85 +95,43 @@ static void vec_destr(igraph_vector_t *vec) {
 
 
 /**
- * \function igraph_random_edge_walk
- * \brief Perform a random walk on a graph and return the traversed edges
- *
- * Performs a random walk with a given length on a graph, from the given
- * start vertex. Edge directions are (potentially) considered, depending on
- * the \p mode argument.
- *
- * \param graph The input graph, it can be directed or undirected.
- *   Multiple edges are respected, so are loop edges.
+ * This function performs a random walk with a given length on a graph,
+ * from the given start vertex.
+ * It's used for igraph_random_walk: 
+ *  - when edge IDs of the traversed edges are needed (weighted or unweighted graphs)
```

Comment:
I would simplify the description here:

```suggestion
 * when weights are used or when edge IDs of the traversed edges are requested.
```

Reply:
Yes, that's actually better. 
```
 * It's used for igraph_random_walk: 
 *    when weights are used or when edge IDs of the traversed edges
 *    and/or vertex IDs of the visited vertices are requested.
```
Does this look fine?